### DISCUSSION TITLE
going modular !python going_modular/train.py question

### DISCUSSION THREAD

**Discussion by RicardoGHa** [mrdbourke/pytorch-deep-learning#1042] [Q&A] [answered]
when I run this command !python going_modular/train.py it doesn't show the process like the visualization of the epochs running :

  0%|                                                     | 0/5 [00:00<?, ?it/s]Epoch: 1 | train_loss: 1.1131 | train_acc: 0.2852 | test_loss: 1.1138 | test_acc: 0.2604
 20%|█████████                                    | 1/5 [00:01<00:04,  1.06s/it]Epoch: 2 | train_loss: 1.0851 | train_acc: 0.4102 | test_loss: 1.1238 | test_acc: 0.1979
 40%|██████████████████                           | 2/5 [00:01<00:02,  1.20it/s]Epoch: 3 | train_loss: 1.0837 | train_acc: 0.4141 | test_loss: 1.1459 | test_acc: 0.1979
 60%|███████████████████████████                  | 3/5 [00:02<00:01,  1.33it/s]Epoch: 4 | train_loss: 1.1104 | train_acc: 0.2930 | test_loss: 1.1318 | test_acc: 0.1979
 80%|████████████████████████████████████         | 4/5 [00:03<00:00,  1.40it/s]Epoch: 5 | train_loss: 1.0833 | train_acc: 0.2930 | test_loss: 1.0883 | test_acc: 0.3712
100%|█████████████████████████████████████████████| 5/5 [00:03<00:00,  1.35it/s]
[INFO] Saving model to: models/05_going_modular_script_mode_tinyvgg_model.pth 

I is also taking so much time to run it has been nearly 18 mins now and it is still running 
![question](https://github.com/user-attachments/assets/6ae9b65d-0a9d-40ee-8c50-77e98b16022d)

**Comment by LuluW8071** [comment]
Run same command but with `!python3` instead of `!python` inside notebook cell

**Reply by RicardoGHa** [reply]
`Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases. ` Got this error message, even tho I have python  
![2](https://github.com/user-attachments/assets/0b7d0c6a-708d-407d-be37-cdd6227f638c)

**Reply by LuluW8071** [reply]
> Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

Its just permission type exception error in windows powershell 
Solution is found [here](https://stackoverflow.com/questions/65348890/python-was-not-found-run-without-arguments-to-install-from-the-microsoft-store)

Did you enter the command shown below or just entered `python3` in notebook??

```bash
!python3 going_modular/train.py
```

You can also do from same thing from bash or any terminal
```bash
python3 going_modular/train.py
```
or
```bash
python going_modular/train.py
```
or
```bash
py going_modular/train.py
```

**Comment by RicardoGHa** [comment]
`Did you enter the command shown below or just entered python3 in notebook??` I entered `!python3 going_modular/train.py` I turned off the python in Manage App Execution Aliases and nothing has changed

**Reply by LuluW8071** [reply, marked as answer]
>  ![image](https://github.com/user-attachments/assets/b84b1131-01aa-48f6-8c92-1646799a84bc)

Looking at ur path structure 
I think the problem is with forward slash, it should be backward slash `going_modular\train.py` in ur case
